src/rdkafka_offset.c

```diff
@@ -400,6 +400,12 @@ rd_kafka_commit(rd_kafka_t *rk,
                 rq   = RD_KAFKA_REPLYQ(repq, 0);
         }
 
+        /* Don't attempt auto commit when rebalancing or initializing since
+         * the rkcg_generation_id is most likely in flux. */
+        if (rkcg->rkcg_subscription &&
```

Comment:
This is not thread safe (this is on the application thread, but rkcg is local to the rdkafka main thread).

Check should probably be performed in rd_kafka_cgrp_offsets_commit().

But we need to make sure that it is ok to commit during the phase of the rebalance when the rebalance callback/event is triggered.

Reply:
Fixed, I also removed the same check from `rd_kafka_cgrp_offset_commit_tmr_cb` since the function will also call `rd_kafka_cgrp_offsets_commit`. I am not sure if this is the ideal behavior for auto commit though as we would no longer "silently" skip committing offsets...